saleor/checkout/views/__init__.py

```diff
@@ -195,6 +195,25 @@ def update_cart_line(request, cart, variant_id):
     return JsonResponse(response, status=status)
 
 
+@get_or_empty_db_cart()
+def clear_cart(request, cart):
+    """Clear cart"""
+    from ..utils import update_cart_quantity
+
+    if not request.is_ajax():
+        return redirect('cart:index')
+
+    lines = cart.lines.all()
+    for line in lines:
```

Comment:
We can iterate over the `cart` object here, as it's `__iter__` method is overridden especially for such purpose 
https://github.com/mirumee/saleor/blob/master/saleor/checkout/models.py#L70-L71